docs/assembly.rst

```diff
@@ -322,7 +323,8 @@ reverts or terminates. As an example, the following assembly snippet is not memo
       revert(0, returndatasize())
     }
 
-But the following is:
+On the other hand, the following code *is* memory safe, because memory beyond the location pointed to by the
+free memory pointed is considered scratch space:
```

Comment:
```suggestion
On the other hand, the following code *is* memory safe, because memory beyond the location pointed to by the
free memory pointer can safely be used as temporary scratch space:
```
Rather maybe?